sdk/core/Azure.Core/tests/TestFramework/ClientTestFixtureAttribute.cs

```diff
@@ -14,13 +14,14 @@ namespace Azure.Core.Testing
     public class ClientTestFixtureAttribute : NUnitAttribute, IFixtureBuilder2, IPreFilter
```

Comment:
What's the point of this change?

Reply:
When `serviceVersions` is empty, `ProcessTest` marks all tests as "SkipRecordings".

Reply:
Hm, but aren't both `serviceVersionNumber` and  `_maxServiceVersion` null in that case?

Reply:
Do you mean that checking `serviceVersion != null` should be enough? Let me check

Reply:
I wonder why current condition doesn't work

Reply:
Current condition uses `int` instead of `int?`, so `serviceVersionNumber` is `0` and `_maxServiceVersion` is `null`.

Reply:
And `Convert.ToInt32` returns `0` for `null`.

Reply:
Ok, I've minimized the change to `ProcessTest` method itself.